### PR TITLE
refactor/cleanup graphdb

### PR DESCRIPTION
**Pull request by iamsmkr**
Rid dead code, restruct project (extracted examples out of docbrown core lib), and refractored code

### CODE REVIEW

File: docbrown/db/src/main.rs [hunk 1]
```diff
@@ -228,24 +167,33 @@ impl TGraphShard {
...
+
+    fn get_shard_num_from_vertex_id(v_id: u64, n_shards: usize) -> usize {
+        let v: usize = v_id.try_into().unwrap();
+        v % n_shards
+    }
+
+    fn parse_record(rec: &StringRecord) -> Option<(u64, u64, i64, u64)> {
+        let src = rec.get(3).and_then(|s| s.parse::<u64>().ok())?;
+        let dst = rec.get(4).and_then(|s| s.parse::<u64>().ok())?;
+        let t = rec.get(5).and_then(|s| s.parse::<i64>().ok())?;
```
Comment (Alnaimi-) [inline docbrown/db/src/main.rs:192]: time?

File: docbrown/db/src/graphdb.rs [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+use rayon::prelude::{IntoParallelRefIterator, ParallelIterator};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct GraphDB {
+    nr_shards: usize,
+    shards: Vec<TemporalGraphPart>,
+}
+
+enum Msg {
```
Comment (fabianmurariu) [inline docbrown/db/src/graphdb.rs:19]: this can be removed

File: examples/btc/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+version.workspace = true
+edition = "2021"
+keywords = ["graph", "temporal-graph", "temporal", "btc"]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+docbrown-core = {path = "../../docbrown/core"}
+docbrown-db = {path = "../../docbrown/db"}
+parking_lot = { version = "0.12" , features = ["serde"] }
```
Comment (fabianmurariu) [inline examples/btc/Cargo.toml:12]: I don't think these all these dependencies are required

File: examples/btc/src/main.rs [hunk 4]
```diff
@@ -10,90 +10,32 @@ use csv::StringRecord;
...
     AddIntoEdge(u64, u64, Vec<(String, Prop)>, i64),
-    Len(futures::channel::oneshot::Sender<usize>),
-    Batch(Vec<Msg>),
-    Done,
+    GetGraphLength(futures::channel::oneshot::Sender<usize>),
+    MsgBatch(Vec<Msg>),
+    FlushMsgs,
 }
 
 struct TGraphShardActor {
```
Comment (fabianmurariu) [inline examples/btc/src/main.rs:41]: very likely we don't need this anymore

File: examples/lotr/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+version.workspace = true
+edition = "2021"
+keywords = ["graph", "temporal-graph", "temporal", "lotr"]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+docbrown-core = {path = "../../docbrown/core"}
+parking_lot = { version = "0.12" , features = ["serde"] }
+flume = "0.10"
```
Comment (fabianmurariu) [inline examples/lotr/Cargo.toml:12]: same as the other one, remove these deps and see what breaks

File: Cargo.toml [hunk 6]
```diff
@@ -1,9 +1,11 @@
 [workspace]
 members = [
 	"docbrown/core",
-	"docbrown/db"
+	"docbrown/db",
+	"examples/btc",
```
Comment (fabianmurariu) [inline Cargo.toml:5]: as we talked, have a look at the examples structure and the version with multiple binaries

File: examples/Cargo.toml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "examples"
+version.workspace = true
+edition = "2021"
+keywords = ["graph", "temporal-graph", "temporal", "examples"]
```
Comment (Alnaimi-) [inline examples/Cargo.toml:5]: add database? @fabianmurariu thoughts

File: examples/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "examples"
+version.workspace = true
+edition = "2021"
```
Comment (Alnaimi-) [inline examples/Cargo.toml:4]: Should be 2023 no?

File: examples/src/bin/btc/main.rs [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    txn: String,
+    amount_btc: u64,
+    _amount_usd: f64,
+    #[serde(with = "custom_date_format")]
+    time: DateTime<Utc>,
+}
+
+#[derive(Deserialize, std::fmt::Debug)]
+pub struct Received {
+    _txn: String,
```
Comment (Alnaimi-) [inline examples/src/bin/btc/main.rs:36]: remove all the starting `_` so variables are standardised

File: examples/src/bin/btc/main.rs [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    collections::hash_map::DefaultHasher,
+    hash::{Hash, Hasher},
+};
+
+#[derive(Deserialize, std::fmt::Debug)]
+pub struct Sent {
+    addr: String,
+    txn: String,
+    amount_btc: u64,
+    _amount_usd: f64,
```
Comment (Alnaimi-) [inline examples/src/bin/btc/main.rs:29]: remove all the starting _ so variables are standardised

Comment (Alnaimi-) [reply examples/src/bin/btc/main.rs:29]: Add TODO / use the field

File: examples/src/bin/btc/main.rs [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+    //    fn serialize<S>(&T, S) -> Result<S::Ok, S::Error>
+    //    where
+    //        S: Serializer
+    //
+    // although it may also be generic over the input types T.
+    pub fn serialize<S>(date: &DateTime<Utc>, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        let s = format!("{}", date.format(FORMAT));
```
Comment (Alnaimi-) [inline examples/src/bin/btc/main.rs:141]: Not sure if we need a `date.unwrap()` here to make sure we fail before the format. Where do we get the initial date object from?

File: examples/Cargo.toml (hunk 8 above)
Comment (iamsmkr) [reply examples/Cargo.toml:4]: That edition = "2021" setting configures your package to be built using the Rust 2021 edition. No further configuration needed!

Refer: https://doc.rust-lang.org/edition-guide/

2021 seems to be latest edition.

File: examples/src/bin/btc/main.rs (hunk 11 above)
Comment (iamsmkr) [reply examples/src/bin/btc/main.rs:141]: taken care by library

**Review by Alnaimi- — APPROVED** [review]

**Pull request merged**